pkg/features/embedded/grub-config/etc/elemental/grub.cfg

```diff
@@ -90,10 +90,10 @@ menuentry "${display_name}" --id active {
 }
 
 for passive_snap in ${passive_snaps}; do
-  menuentry "${display_name} (snapshot ${passive_snap})" --id ${passive_snap} {
+  menuentry "${display_name} (snapshot ${passive_snap})" --id passive${passive_snap} ${passive_snap} {
     set mode=passive
     search --no-floppy --label --set=root ${state_label}
-    set_volume ${passive_snap}
```

Comment:
This variable inside the menu is not expanded when looping, in fact it is expanded when choosing the actual menu item and it takes the value of the last assignement in the loop. Should have been passed as an argument to the `menuentry` call.